jetty-websocket/websocket-core/src/main/java/org/eclipse/jetty/websocket/core/internal/WebSocketCoreSession.java

```diff
@@ -570,6 +571,8 @@ public boolean isAutoFragment()
     @Override
     public void setAutoFragment(boolean autoFragment)
     {
+        if (autoFragment && getExtensionStack().getRsv1User() instanceof DeflateFrameExtension)
```

Comment:
Can you add a TODO here say that if we get more examples of checks like this, then we need an extensible/generic mechanism for extensions to validate changes to configuration.